extractor/filesystem/os/opkg/opkg_test.go

```diff
@@ -0,0 +1,844 @@
+	"github.com/google/osv-scalibr/extractor/filesystem/internal/units"
+	"github.com/google/osv-scalibr/extractor/filesystem/os/opkg"
+	"github.com/google/osv-scalibr/extractor/filesystem/simplefileapi"
+	scalibrfs "github.com/google/osv-scalibr/fs"
+	scalibrlog "github.com/google/osv-scalibr/log"
+	"github.com/google/osv-scalibr/purl"
+	"github.com/google/osv-scalibr/stats"
+	"github.com/google/osv-scalibr/testing/fakefs"
+	"github.com/google/osv-scalibr/testing/testcollector"
+)
+
+func TestNew(t *testing.T) {
+	tests := []struct {
+		name    string
+		cfg     opkg.Config
+		wantCfg opkg.Config
+	}{
+		{
+			name: "default",
+			cfg:  opkg.DefaultConfig(),
+			wantCfg: opkg.Config{
+				MaxFileSizeBytes:    10 * units.MiB,
+				IncludeNotInstalled: false,
+			},
+		},
+		{
+			name: "custom",
+			cfg: opkg.Config{
+				MaxFileSizeBytes:    10,
+				IncludeNotInstalled: true,
+			},
+			wantCfg: opkg.Config{
+				MaxFileSizeBytes:    10,
+				IncludeNotInstalled: true,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := opkg.New(tt.cfg)
+			if !reflect.DeepEqual(got.Config(), tt.wantCfg) {
+				t.Errorf("New(%+v).Config(): got %+v, want %+v", tt.cfg, got.Config(), tt.wantCfg)
+			}
+		})
+	}
+}
+
+func TestFileRequired(t *testing.T) {
+	tests := []struct {
+		name             string
+		path             string
+		fileSizeBytes    int64
+		maxFileSizeBytes int64
+		wantRequired     bool
+		wantResultMetric stats.FileRequiredResult
+	}{
+		{
+			name:             "status file",
+			path:             "usr/lib/opkg/status",
+			wantRequired:     true,
+			wantResultMetric: stats.FileRequiredResultOK,
+		},
+		{
+			name:         "status as a directory",
+			path:         "usr/lib/opkg/status/foo",
+			wantRequired: false,
+		},
+		{
+			name:             "status file required if file size < max file size",
+			path:             "usr/lib/opkg/status",
+			fileSizeBytes:    10 * units.KiB,
+			maxFileSizeBytes: 100 * units.KiB,
+			wantRequired:     true,
+			wantResultMetric: stats.FileRequiredResultOK,
+		},
+		{
+			name:             "status file required if file size == max file size",
+			path:             "usr/lib/opkg/status",
+			fileSizeBytes:    100 * units.KiB,
+			maxFileSizeBytes: 100 * units.KiB,
+			wantRequired:     true,
+			wantResultMetric: stats.FileRequiredResultOK,
+		},
+		{
+			name:             "status file not required if file size > max file size",
+			path:             "usr/lib/opkg/status",
+			fileSizeBytes:    1000 * units.KiB,
+			maxFileSizeBytes: 100 * units.KiB,
+			wantRequired:     false,
+			wantResultMetric: stats.FileRequiredResultSizeLimitExceeded,
+		},
+		{
+			name:             "status file required if max file size set to 0",
+			path:             "usr/lib/opkg/status",
+			fileSizeBytes:    100 * units.KiB,
+			maxFileSizeBytes: 0,
+			wantRequired:     true,
+			wantResultMetric: stats.FileRequiredResultOK,
+		},
```

Comment:
Add a case where FileRequired is false due to the file path not matching the expected prefix.